tests/scripts/thread-cert/Cert_5_1_09_REEDAttachConnectivity.py

```diff
@@ -94,65 +121,161 @@ def test(self):
+        #            - Scan Mask TLV
+        #                If the DUT sends multiple MLE Parent Requests
+        #                    - The first one MUST be sent only to all Routers
+        #                    - Subsequent ones MAY be sent to all Routers and REEDS
+        #            -  Version TLV
+        #         If the first MLE Parent Request was sent to all Routers and
+        #         REEDS, the test fails.
+
+        pkts_copy = pkts.copy()
+        pkts.filter_wpan_src64(ROUTER_2).\
+            filter_LLARMA().\
+            filter_mle_cmd(MLE_PARENT_REQUEST).\
+            filter(lambda p: {
+                              CHALLENGE_TLV,
+                              MODE_TLV,
+                              SCAN_MASK_TLV,
+                              VERSION_TLV
+                              } <= set(p.mle.tlv.type) and\
+                   p.ipv6.hlim == 255 and\
+                   p.mle.tlv.scan_mask.r == 1 and\
+                   p.mle.tlv.scan_mask.e == 0
+                   ).\
+            must_next()
+        lstart = pkts.index
+
+        # Step 5: The DUT MUST send a MLE Parent Request with an IP hop limit of
+        #         255 to the Link-Local All Routers multicast address (FF02::2).
+        #         The following TLVs MUST be present in the MLE Parent Request:
+        #            - Challenge TLV
+        #            - Mode TLV
+        #            - Scan Mask TLV
+        #               - The Scan Mask TLV MUST be sent to Routers And REEDs
+        #            -  Version TLV
+        #
+        #         In securing the first three messages of the attaching process,
+        #         the full four-byte key sequence number MUST be included in
+        #         the Auxiliary Security Header used for MLE security.
+        #
+        #         To send the full four-byte key sequence number, the Key
+        #         Identifier Mode of the Security Control Field SHALL be set to
+        #         ‘0x02’, indicating the presence of a four-byte Key Source,
+        #         which SHALL contain the four-byte key sequence number in
+        #         network byte order.
+
+        pkts.filter_wpan_src64(ROUTER_2).\
+            filter_LLARMA().\
+            filter_mle_cmd(MLE_PARENT_REQUEST).\
+            filter(lambda p: {
+                              CHALLENGE_TLV,
+                              MODE_TLV,
+                              SCAN_MASK_TLV,
+                              VERSION_TLV
+                              } <= set(p.mle.tlv.type) and\
+                   p.ipv6.hlim == 255 and\
+                   p.mle.tlv.scan_mask.r == 1 and\
+                   p.mle.tlv.scan_mask.e == 1 and\
+                   p.wpan.aux_sec.key_id_mode == 0x2
+                   ).\
+            must_next()
+        lend = pkts.index
+
+        # Step 4: REED_1 and REED_2 no response to Parent Request meant for all routers.
+
+        for i in range(1, 3):
+            pkts_copy.range(lstart, lend).\
```

Comment:
We can just use `pkts` since `must_not_next()` never change the current index of `pkts`.
```suggestion
            pkts.range(lstart, lend).\
```